code/modules/mod/modules/modules_maint.dm

```diff
@@ -20,22 +23,37 @@
 
 /obj/item/mod/module/springlock/on_suit_activation()
 	RegisterSignal(mod.wearer, COMSIG_ATOM_EXPOSE_REAGENTS, PROC_REF(on_wearer_exposed))
+	AddComponent(/datum/component/connect_loc_behalf, mod.wearer, gas_connections)
 
 /obj/item/mod/module/springlock/on_suit_deactivation(deleting = FALSE)
 	UnregisterSignal(mod.wearer, COMSIG_ATOM_EXPOSE_REAGENTS)
+	qdel(GetComponent(/datum/component/connect_loc_behalf))
 
-///Signal fired when wearer is exposed to reagents
-/obj/item/mod/module/springlock/proc/on_wearer_exposed(atom/source, list/reagents, datum/reagents/source_reagents, methods, volume_modifier, show_message)
-	SIGNAL_HANDLER
-
-	if(!(methods & (VAPOR|PATCH|TOUCH)) || set_off || mod.wearer.stat == DEAD)
-		return //remove non-touch reagent exposure
+///Fires a signal to snap shut the springlock when called
```

Comment:
"Fires a signal"? Doesn't seem to

Reply:
Thanks for pointing stuff out. I'll check everything over and fix it up when I have time later.